### PR TITLE
Upgrade to Jupyter Collaboration 3

### PR DESCRIPTION
**Pull request by brichet**
Update dependencies:
- `jupyter_collaboration >= 3` (python and javascript)
- `@jupyter/ydoc>2` (javascript)

### CODE REVIEW

**Comment by dlqqq** [comment]
Motivation for this change (authored by @brichet in a private conversation):

It is related to https://github.com/jupyterlab/jupyter-collaboration/pull/379, that should be in soon, and adds a widget to display the current users of a collaborative document.
IIRC this PR can't be backported to jupyter-collaboration 2 because of its dependencies to jupyter/ydoc .

**Review by dlqqq — CHANGES_REQUESTED** [review]
@brichet Thank you for working on this while I was away! Left a few points of feedback below.

**Comment by dlqqq** [comment]
BTW you may have to close & re-open this PR to re-run the workflows. I noticed the readthedocs workflows stalled for PRs opened last week, but the issue should be fixed now.

File: packages/jupyterlab-chat/package.json [hunk 1]
```diff
@@ -52,8 +52,8 @@
   },
   "dependencies": {
     "@jupyter/chat": "^0.6.2",
-    "@jupyter/docprovider": "^2.1.4",
-    "@jupyter/ydoc": "^1.1.1",
+    "@jupyter/collaborative-drive": "^3.0.0",
+    "@jupyter/ydoc": "^2.0.0 || ^3.0.0",
```
[the remark this answers is not in the document]
Comment (brichet) [reply packages/jupyterlab-chat/package.json:56]: It is just to get aligned with `@jupyter/collaborative-drive` dependencies, which support both.
We can probably drop support for version 2.

File: packages/jupyterlab-chat/src/ychat.ts [hunk 2]
```diff
@@ -96,6 +96,37 @@ export class YChat extends YDocument<IChatChanges> {
...
 
+  getSource(): JSONObject {
+    const users = this._users.toJSON();
+    const messages = this._messages.toJSON();
+    const metadata = this._metadata.toJSON();
+
+    return { users, messages, metadata };
+  }
+
+  setSource(value: JSONObject): void {
```
[the remark this answers is not in the document]
Comment (brichet) [reply packages/jupyterlab-chat/src/ychat.ts:107]: It is required in `@jupyter/ydoc` >= 3
https://github.com/jupyter-server/jupyter_ydoc/blob/43532219006844c16dcccb4f9831daf7ef11995e/javascript/src/ydocument.ts#L171

File: packages/jupyterlab-chat/src/ychat.ts [hunk 3]
```diff
@@ -96,6 +96,37 @@ export class YChat extends YDocument<IChatChanges> {
     return JSONExt.deepCopy(this._messages.toJSON());
   }
 
+  getSource(): JSONObject {
```
[the remark this answers is not in the document]
Comment (brichet) [reply packages/jupyterlab-chat/src/ychat.ts:99]: I don't think, it must be `JSONValue | string`, as described in the `YDocument` https://github.com/jupyter-server/jupyter_ydoc/blob/43532219006844c16dcccb4f9831daf7ef11995e/javascript/src/ydocument.ts#L171

**Comment by brichet** [comment]
Bump CI

**Review by dlqqq — APPROVED** [review]
Awesome, TY for answering my feedback so thoughtfully!

**Pull request merged**